api/cas/service.py

```diff
@@ -0,0 +1,106 @@
+from rest_framework.exceptions import ValidationError, PermissionDenied
+
+from api.cas import messages
+
+from osf.models import ApiOAuth2Application, ApiOAuth2PersonalToken, ApiOAuth2Scope, Institution, OSFUser
+
+
+def get_oauth_token(body_data):
```

Comment:
The four methods in this class are all ported from CAS. Please refer to the doc.